src/docfx/build/xref/XrefMap.cs

```diff
@@ -19,21 +19,21 @@ internal class XrefMap
         private readonly IReadOnlyDictionary<string, XrefSpec> _externalXrefMap;
         private readonly Context _context;
 
-        private static ThreadLocal<Stack<Document>> t_recursionDetector = new ThreadLocal<Stack<Document>>(() => new Stack<Document>());
+        private static ThreadLocal<Stack<(string propertyName, Document parent)>> t_recursionDetector = new ThreadLocal<Stack<(string, Document)>>(() => new Stack<(string, Document)>());
```

Comment:
does it need `uid` for completeness? what if a document has multiple xrefspecs (uids)

Reply:
We only support uid definition on root level for SDP now, will it be multiple uids later?

Reply:
Add `uid` to the key, no test case for it since we can couldn't define multiple `uid` for now